deps/oob-1.0.2/jsonutil.c

```diff
@@ -404,7 +530,84 @@ void setoobConnectioninfo(const char* host,
   else
     copyString("http", connectionInfo.protocol, 8);
+}
 
+void setoobdnsinfo(const char *key, const char *val) {
+    char* caps = malloc(strlen(key));
+    upper(caps, key);
+
+    if (!strcmp(caps, "DESCRIPTION")) {
+        copyString(key, dnsKeys.description, 16);
+        copyString(val, dnsValues.description, 4096);
+    } else if (!strcmp(caps, "DRIVER")) {
+        copyString(key, dnsKeys.driver, 8);
+        copyString(val, dnsValues.driver, 64);
+    } else if (!strcmp(caps, "LOCALE")) {
+        copyString(key, dnsKeys.locale, 8);
+        copyString(val, dnsValues.locale, 64);
+    } else if (!strcmp(caps, "SERVER")) {
+        copyString(key, dnsKeys.host, 8);
+        copyString(val, dnsValues.host, 512);
+    } else if (!strcmp(caps, "PORT")) {
+        copyString(key, dnsKeys.port, 8);
+        copyString(val, dnsValues.port, 10);
+    } else if (!strcmp(caps, "ACCOUNT")) {
+        copyString(key, dnsKeys.account, 8);
+        copyString(val, dnsValues.account, 256);
+    } else if (!strcmp(caps, "UID")) {
+        copyString(key, dnsKeys.user, 8);
+        copyString(val, dnsValues.user, 256);
+    } else if (!strcmp(caps, "PWD")) {
+        copyString(key, dnsKeys.password, 8);
+        copyString(val, dnsValues.password, 8);
+    } else if (!strcmp(caps, "TOKEN")) {
+        copyString(key, dnsKeys.token, 8);
+        copyString(val, dnsValues.token, 1024);
+    } else if (!strcmp(caps, "AUTHENTICATOR")) {
+        copyString(key, dnsKeys.authenticator, 16);
+        copyString(val, dnsValues.authenticator, 1024);
+    } else if (!strcmp(caps, "DATABASE")) {
+        copyString(key, dnsKeys.database, 16);
+        copyString(val, dnsValues.database, 256);
+    } else if (!strcmp(caps, "SCHEMA")) {
+        copyString(key, dnsKeys.schema, 8);
+        copyString(val, dnsValues.schema, 256);
+    } else if (!strcmp(caps, "WAREHOUSE")) {
+        copyString(key, dnsKeys.warehouse, 16);
+        copyString(val, dnsValues.warehouse, 256);
+    } else if (!strcmp(caps, "ROLE")) {
+        copyString(key, dnsKeys.role, 8);
+        copyString(val, dnsValues.role, 256);
+    } else if (!strcmp(caps, "SSL")) {
+        copyString(key, dnsKeys.ssl, 8);
+        copyString(val, dnsValues.ssl, 10);
+    } else if (!strcmp(caps, "TRACING")) {
+        copyString(key, dnsKeys.tracing, 8);
+        copyString(val, dnsValues.tracing, 10);
+    } else if (!strcmp(caps, "JWT_TIME_OUT")) {
+        copyString(key, dnsKeys.jwtTimeout, 16);
+        copyString(val, dnsValues.jwtTimeout, 256);
+    } else if (!strcmp(caps, "TIMEZONE")) {
+        copyString(key, dnsKeys.timezone, 16);
+        copyString(val, dnsValues.timezone, 64);
+    } else if (!strcmp(caps, "PRIV_KEY_FILE")) {
+        copyString(key, dnsKeys.privKey, 16);
+        copyString(val, dnsValues.privKey, 1024);
+    } else {
+        free(caps);
+        return;
+    }
+    free(caps);
+    return;
+}
+
+void upper(char* caps, const char* word){
```

Comment:
You don't need this, and you should never have to write this.

Reply:
We do because you wanted case sensitive keys and the only way to consolidate the difference between something like "server" and "SERVER".

Reply:
Use stricmp or strcasecmp instead. As a rule you should never be writing such basic string processing functions.